app/src/main/java/com/concordia/mcga/helperClasses/OutdoorDirection.java

```diff
@@ -0,0 +1,176 @@
+package com.concordia.mcga.helperClasses;
+
+import android.content.Context;
+
+import com.akexorcist.googledirection.DirectionCallback;
+import com.akexorcist.googledirection.GoogleDirection;
+import com.akexorcist.googledirection.constant.TransportMode;
+import com.akexorcist.googledirection.constant.Unit;
+import com.akexorcist.googledirection.model.Direction;
+import com.akexorcist.googledirection.model.Leg;
+import com.akexorcist.googledirection.model.Route;
+import com.akexorcist.googledirection.model.Step;
+import com.akexorcist.googledirection.util.DirectionConverter;
+import com.concordia.mcga.activities.R;
+import com.concordia.mcga.exceptions.MCGAPathFindingException;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Polyline;
+import com.google.android.gms.maps.model.PolylineOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+class OutdoorDirection implements DirectionCallback {
+    private String serverKey;
+    private final int transitPathWidth = 5;
+    private final int transitPathColor = 0x80ed1026; // transparent red
+    private final int walkingPathWidth = 3;
+    private final int walkingPathColor = 0x801767e8; // transparent blue
+    private LatLng origin, destination;
+    private List<Polyline> polylines;
+    private List<Step> steps;
+    private List<String> instructions;
+    private Leg leg;
+    private GoogleMap map;
+    private Context context;
+    private String transportMode = TransportMode.BICYCLING;
+
+    public OutdoorDirection() {
+        polylines = new ArrayList<>();
+        steps = new ArrayList<>();
+        instructions = new ArrayList<>();
+        transportMode = TransportMode.TRANSIT; // default transport mode
+    }
+
+    /**
+     * Upon successful response from Google Direction Server, Create a path between the origin and destion
+     *
+     * @param direction raw directions in JSON format are wrapped with the Direction class for ease of use
+     * @param rawBody   raw directions in JSON format
+     */
+    @Override
+    public void onDirectionSuccess(Direction direction, String rawBody) {
+        if (direction.isOK()) {
+            Route route = direction.getRouteList().get(0);
+            leg = route.getLegList().get(0);
+            steps = leg.getStepList();
+        }
+    }
+
+    @Override
+    public void onDirectionFailure(Throwable t) {
+        try {
+            throw new MCGAPathFindingException(t.getMessage());
+        } catch (MCGAPathFindingException e) {
+            e.printStackTrace();
```

Comment:
Because i cannot throw an exception since I'm overriding a callback method of the external library. Therefor "throws" cannot be added to the method definition. However I can just print the stack trace. Will fix.